internal/lock.go

```diff
@@ -149,11 +149,21 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	status_line := Status_Line{filteredResources, make(chan int)}
```

Comment:
You should have a method named `NewStatusLine(filteredResources)` that builds the status line. This way the call sites don't need to know about the internals of the status line.

Reply:
Should the values be passed to this NewStatusLine constructor, then passed to run(), then passed to composeLine()? Or should NewStatusLine store them as fields of the StatusLine struct and run() composeLine() can just access them directly from the pointer/object?

Reply:
Would it be better for the variables/data to be passed from one function to another (NewStatusLine to run to composeLine) or for them to be stored in StatusLine struct fields that each method can directly access via pointer/object?

Reply:
If you're talking about init data that will be used for the entire life of the object, they should be stored as field, yes.